src/ext/php5_4/engine_hooks.c

```diff
@@ -670,3 +670,17 @@ void ddtrace_execute_internal_mshutdown(void) {
         dd_prev_execute_internal = dd_prev_execute_internal == execute_internal ? NULL : dd_prev_execute_internal;
     }
 }
+
+// TODO: can we support close-at-exit and by extension fatal errors on PHP 5.4?
+zval *ddtrace_make_exception_from_error(DDTRACE_ERROR_CB_PARAMETERS TSRMLS_DC) {
+    PHP5_UNUSED(DDTRACE_ERROR_CB_PARAM_PASSTHRU TSRMLS_CC);
+
+    return NULL;
+}
+
+void ddtrace_close_all_open_spans(TSRMLS_D) {
+#if ZTS
+    PHP5_UNUSED(TSRMLS_C);
+#endif
+    ddtrace_log_debug("Request to close all open spans ignored; not supported on PHP 5.4 (yet, anyway)");
```

Comment:
Should we still free the span stacks at this point?

Reply:
I don't think so; the C stack will be holding onto variables that would be invalidated if we did that.